pkg/apis/policy/types.go

```diff
@@ -215,6 +208,10 @@ type PodSecurityPolicySpec struct {
 	// is allowed in the "Volumes" field.
 	// +optional
 	AllowedFlexVolumes []AllowedFlexVolume
+	// Sysctls is a white list of allowed sysctls in a pod spec. Each entry
+	// is either a plain sysctl name or ends in "*" in which case it is considered
+	// as a prefix of allowed sysctls. Empty list means no sysctls allowed.
+	Sysctls []string
```

Comment:
why is this a []string and not like in the internal types a `[]Sysctl`? The later allow us extensions later. Also we could have a different default policy for safe and unsafe sysctls.

Reply:
We have only one default policy for safe sysctl which is allow all. In other cases all allowed sysctls (safe and unsafe) must be specified explicitly. There is no way to specify (not even the default behaviour) allowing all unsafe sysctls to be set. For that reason there is no need to distinguish between safe and unsafe sysctls on the PSP level now.

Though, yes. If we decide to add more policies, we will need to replace `[]string` with something more complex, e.g. `[]Sysctl`, which will be then backward incompatible. Given we are about to promote to Beta, it would be hard to violate the guarantee of the sysctl feature API to be stable and backward compatible with the changes.

Reply:
In the second PR (https://github.com/kubernetes/kubernetes/pull/64264/commits/6ed9c900190b6177177f56099fb66dbfe5013535#diff-742ab7cab01feaff825dd14a18295532R217) I have created:
```go
type PodDisruptionBudgetSpec struct {
	...
	Sysctls *SysctlsWhitelist
}
// SysctlsWhitelist is a whitelist of allowed sysctls in a pod spec
type SysctlsWhitelist struct {
	// Items is a whitelist of allowed sysctls in a pod spec. Each entry is either a plain sysctl name
	// or ends in "*" in which case it is considered as a prefix of allowed sysctls.
	// +optional
	Items []string
}
```

So we could put all the new policies into the `SysctlsWhitelist` struct. The PSP owner should than do:

```yaml
apiVersion: v1
kind: PodSecurityPolicy
spec:
- sysctls:
    items:
    - kernel.shmmax
    - kernel.shmall
    unsafePolicy:
    ...
    safePolicy:
    ... 
```

to get more granular restriction of the whitelist.